### DISCUSSION TITLE
Can't connect to multiple Auth0 account (next-auth always use username-password auth)

### DISCUSSION THREAD

**Discussion by MatteoGauthier** [nextauthjs/next-auth#7884] [Help]
Hey ! Hope you are doing well, I have an issue while using next-auth with Auth0. When I try to connect to AuthO using a Social Connections (Google), next-auth always connect me with the wrong account.

Here is the scenario : 
1. Go to /api/auth/signin/auth0
2. I'm beeing redirected to Auth0 login page
3. I create an account with the email `lorem-ipsum@gmail.com`
4. I'm now authenticated and redirected to my app
5. Now I'm disconecting with /api/auth/signout
6. I don't have a connected session now
7. I go to /api/auth/signin/auth0 in order to connect another account
8. On the Auth0 page I choose "Continue with Google"
9. I choose my Google account with a different email adress (`foo-bar@gmail.com`)
10. Cool, but when the authentification succeed I'm now connected with the `lorem-ipsum@gmail.com` account ?? (I impersonate the `lorem-ipsum@gmail.com` account)

Here is my next-auth config :
```ts
export const authOptions: NextAuthOptions = {
  callbacks: {
    session: ({ session, user }) => {
      console.log("session", session)
      console.log("user", session)

      return {
        ...session,
        user: {
          ...session.user,
          id: user.id,
          role: user.role,
          currentCampaignId: user.currentCampaignId,
        },
      }
    },
  },
  adapter: PrismaAdapter(prisma),
  debug: true,
  secret: env.NEXTAUTH_SECRET,
  providers: [
    Auth0Provider({
      clientId: env.AUTH0_CLIENT_ID,
      clientSecret: env.AUTH0_CLIENT_SECRET,
      issuer: env.AUTH0_ISSUER,
      authorization: {
        params: {
          prompt: "login",
        },
      },
      profile(profile) {
        return {
          id: profile.sub,
          name: profile.name,
          email: profile.email,
          image: profile.picture,
          role: "USER",
        }
      },
    }),
  ],
}
```

I really don't understand what is happening here 🤔

I've enable next-auth debug logs, here it is : 
```
[next-auth][debug][OAUTH_CALLBACK_RESPONSE] {
  profile: {
    id: 'google-oauth2|<foo-bar@gmail.com id>',
    name: 'Mattèo Gauthier',
    email: 'foo-bar@gmail.com',
    image: 'https://lh3.googleusercontent.com/a/AAcHTtcp3llzNeWYsLEqEy-OXa8-m6nLGnNHrIKmyfv-covHtKjx=s96-c',
    role: 'USER'
  },
  account: {
    provider: 'auth0',
    type: 'oauth',
    providerAccountId: 'google-oauth2|<foo-bar@gmail.com id>',
    access_token: 'xxxx',
    id_token: 'xxxx',
    scope: 'openid profile email',
    expires_at: 1687966798,
    token_type: 'Bearer'
  },
  OAuthProfile: {
    given_name: 'Mattèo',
    family_name: 'Gauthier',
    nickname: 'matteo.gauthier',
    name: 'Mattèo Gauthier',
    picture: 'https://lh3.googleusercontent.com/a/AAcHTtcp3llzNeWYsLEqEy-OXa8-m6nLGnNHrIKmyfv-covHtKjx=s96-c',
    locale: 'fr',
    updated_at: '2023-06-27T15:39:58.164Z',
    email: 'foo-bar@gmail.com',
    email_verified: true,
    iss: 'https://xxxxx.eu.auth0.com/',
    aud: 'xxxx',
    iat: 1687880398,
    exp: 1687916398,
    sub: 'google-oauth2|<foo-bar@gmail.com id>',
    auth_time: 1687880398,
    sid: 'xx-xxx'
  }
}
[next-auth][debug][adapter_getUserByAccount] {
  args: [
    {
      providerAccountId: 'google-oauth2|<foo-bar@gmail.com id>',
      provider: 'auth0'
    }
  ]
}
[next-auth][debug][adapter_getUserByAccount] {
  args: [
    {
      providerAccountId: 'google-oauth2|<foo-bar@gmail.com id>',
      provider: 'auth0'
    }
  ]
}
[next-auth][debug][adapter_createSession] {
  args: [
    {
      sessionToken: '72b1e922-df55-40d7-b82f-89da11e656bc',
      userId: 'cljee4am10000l908me3vcax1',
      // ⚠️⚠️⚠️ Wrong ID ⬆️⬆️ for foo-bar@gmail.com, here it's the id of lorem-ipsum@gmail.com ⚠️⚠️⚠️
      expires: 2023-07-27T15:39:59.439Z
    }
  ]
}
```

In my PostgreSQL Database (Prisma) I also see the issue
<img width="801" alt="SCR-20230627-pslc" src="https://github.com/nextauthjs/next-auth/assets/32040951/f742a77d-041c-4138-9e40-f890ab2014b8">

Has anyone ever had this problem or could it be solved?

**Comment by justapithecus** [comment]
@MatteoGauthier Did you ever determine the root cause of this?

I'm encountering a similar issue: https://github.com/nextauthjs/next-auth/discussions/8378. It seems like this is the default behavior when using the Prisma adapter? It is incorrectly linking entirely different provider/accounts to the same user, and only creating a single user for the very first sign in (which seems to not be related to accounts and users, but related to browser cache). And in your screenshot, it also incorrectly labeled the google provider as `auth0` as well.

My current workaround is to unfortunately avoid using the prisma adapter entirely until I can get some guidance from the community.
